apps/designer/app/canvas/shared/sync/use-sync.ts

```diff
@@ -14,23 +15,34 @@ export const useSync = ({
 }) => {
   useInterval(() => {
     const entries = sync();
-    if (entries.length === 0) {
+    const messages = syncMessages();
```

Comment:
In case of `fetch` fail seems like messages used here will be lost.
Also using `useInterval` gives 0 order guarantee. I mean that 1 fetch can "patch" data later when 2nd.

IMO all such systems must not do any operations until the response from the previous has been received.

Reply:
Here I just mimic existing code. Agree not ideal. Though we need first figure out how to bypass current json limitations so we could do more atomic updates and get rid of this for good.